bee-node/src/plugins/version_checker.rs

```diff
@@ -19,9 +23,14 @@ pub struct VersionChecker {}
 #[async_trait]
 impl<N: Node> Worker<N> for VersionChecker {
     type Config = ();
-    type Error = Infallible;
+    type Error = reqwest::Error;
```

Comment:
I suggest creating an `Error` enum:
```Rust
enum VersionCheckerError {
    RequestError(request::Error),
    AlreadyLatestVersion,
}
```
to have more self-documenting code instead of using a boolean to indicate when already up-to-date.

Reply:
The node already being up-to-date is not a worker error though

Reply:
So you would remove the `AlreadyLatestVersion` from the error enum?

Reply:
Yes, I don't see when this would be returned by the actor.